statusHandler/termuic/termuiRenders/widgetsRender.go

```diff
@@ -0,0 +1,217 @@
+	wr.pShardId = widgets.NewParagraph()
+
+	wr.pCountLeader = widgets.NewParagraph()
+	wr.pCountConsensus = widgets.NewParagraph()
+	wr.pCountAcceptedBlocks = widgets.NewParagraph()
+
+	wr.pTxPoolLoad = widgets.NewParagraph()
+	wr.pNumConnectedPeers = widgets.NewParagraph()
+
+	wr.lLog = widgets.NewList()
+
+	wr.tSyncInfo = widgets.NewTable()
+	wr.tSyncInfo.Rows = [][]string{
+		{"", "", "", ""},
+	}
+
+	wr.lInstanceInfo = widgets.NewList()
+	wr.lChainInfo = widgets.NewList()
+}
+
+func (wr *WidgetsRender) setGrid() {
+	wr.grid.Set(
+		ui.NewRow(completRow/7, wr.pPublicKey),
+		ui.NewRow(completRow/10, wr.tSyncInfo),
+		ui.NewRow(completRow/3,
+			ui.NewCol(completCol/2,
+				ui.NewRow(completRow/4, wr.pShardId),
+				ui.NewRow(completRow/4, wr.pCountConsensus),
+				ui.NewRow(completRow/4, wr.pCountLeader),
+				ui.NewRow(completRow/4, wr.pCountAcceptedBlocks),
+			),
+			ui.NewCol(completCol/2,
+				ui.NewRow(completRow/2, wr.pTxPoolLoad),
+				ui.NewRow(completRow/2, wr.pNumConnectedPeers),
+			),
+		),
+		ui.NewRow(completRow/3, wr.lLog),
+	)
+}
+
+//RefreshData method is used to prepare data that are displayed on grid
+func (wr *WidgetsRender) RefreshData(logLines []string) {
+	nonceI, _ := wr.termuiConsoleMetrics.Load(core.MetricNonce)
+	nonce := nonceI.(int)
+
+	synchronizedRoundI, _ := wr.termuiConsoleMetrics.Load(core.MetricSynchronizedRound)
+	synchronizedRound := synchronizedRoundI.(int)
+
+	currentRoundI, _ := wr.termuiConsoleMetrics.Load(core.MetricCurrentRound)
+	currentRound := currentRoundI.(int)
+
+	isSyncingI, _ := wr.termuiConsoleMetrics.Load(core.MetricIsSyncing)
+	isSyncing := isSyncingI.(int)
+
+	wr.prepareSyncInfoForDisplay(nonce, currentRound, synchronizedRound, isSyncing)
+
+	publicKeyI, _ := wr.termuiConsoleMetrics.Load(core.MetricPublicKey)
+	publicKey := publicKeyI.(string)
+	wr.preparePublicKeyForDisplay(publicKey)
+
+	shardIdI, _ := wr.termuiConsoleMetrics.Load(core.MetricShardId)
+	shardId := shardIdI.(int)
+	wr.prepareShardIdForDisplay(shardId)
+
+	numConnectedPeersI, _ := wr.termuiConsoleMetrics.Load(core.MetricNumConnectedPeers)
+	numConnectedPeers := numConnectedPeersI.(int)
+
+	txPoolLoadI, _ := wr.termuiConsoleMetrics.Load(core.MetricTxPoolLoad)
+	txPoolLoad := txPoolLoadI.(int)
+	wr.prepareTxPoolLoadForDisplay(txPoolLoad)
+	wr.prepareNumConnectedPeersForDisplay(numConnectedPeers)
+
+	countConsensusI, _ := wr.termuiConsoleMetrics.Load(core.MetricCountConsensus)
+	countConsensus := countConsensusI.(int)
+
+	countLeaderI, _ := wr.termuiConsoleMetrics.Load(core.MetricCountLeader)
+	countLeader := countLeaderI.(int)
+
+	countAcceptedBlocksI, _ := wr.termuiConsoleMetrics.Load(core.MetricCountAcceptedBlocks)
+	countAcceptedBlocks := countAcceptedBlocksI.(int)
+
+	wr.prepareConsensusInformationsForDisplay(countConsensus, countLeader, countAcceptedBlocks)
+
+	wr.prepareListWithLogsForDisplay(logLines)
+
+	return
+}
+
+func (wr *WidgetsRender) prepareListWithLogsForDisplay(logData []string) {
```

Comment:
Duplicate code with the logRender. Try embed the logRender in widgetRendr or use the same code.

Reply:
Refactor on next pull request.